src/OrchardCore.Modules/OrchardCore.Users/Deployment/LoginSettingsDeploymentSource.cs

```diff
@@ -0,0 +1,38 @@
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Settings;
+using OrchardCore.Users.Services;
+
+namespace OrchardCore.Users.Deployment
+{
+    public class LoginSettingsDeploymentSource : IDeploymentSource
+    {
+        private readonly IMembershipService _membershipService;
+        private readonly ISiteService _siteService;
+
+        public LoginSettingsDeploymentSource(IMembershipService membershipService, ISiteService siteService)
+        {
+            _membershipService = membershipService;
+            _siteService = siteService;
+        }
+
+        public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+        {
+            var loginSettingsStep = step as LoginSettingsDeploymentStep;
+
+            if (loginSettingsStep == null)
+            {
+                return;
+            }
+
+            var loginSettings = await _membershipService.GetLoginSettingsAsync();
```

Comment:
I don't think you need to move this to a service. Just use `ISite` directly here. (and if it is null, just don't add it because it would not need to be there anyway)

But this one has no security implications that I can see, so if you agree we can change that, and go ahead with it

Reply:
Like this?
`var loginSettings = (await _siteService.GetSiteSettingsAsync()).As<LoginSettings>();`